update zlib and add unittest case

TimeYWL opened the pull request:
<!-- TemplateReference: https://github.com/PaddlePaddle/Paddle/wiki/PULL-REQUEST-TEMPLATE--REFERENCE -->
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->

### PR Category
<!-- One of [ User Experience | Execute Infrastructure | Operator Mechanism | CINN | Custom Device | Performance Optimization | Distributed Strategy | Parameter Server | Communication Library | Auto Parallel | Inference | Environment Adaptation | Others ] -->
Others 

### PR Types
<!-- One of [ New features | Bug fixes | Improvements | Performance | BC Breaking | Deprecations | Docs | Devs | Not User Facing | Security | Deprecations | Others ] -->
Bug fixes

### Description
<!-- Describe what you’ve done -->
1.Compiling paddle whith third_part/zlib(1.2.8) will cause to SF bug on Kylin OS zlib(1.2.11).Reference to https://github.com/PyTables/PyTables/issues/686, I updating third_party/zlib version to 1.2.11 to avoid SF bug successfully.
2.Add _C_ops.full_() operator test cases which contain fp16, fp32, fp64 and bf16 types, that was used on llama project.

tc20042008 committed:
FLAGS_logging_pir_py_code (#63981)

* FLAGS_logging_pir_py_code

* FLAGS_logging_pir_py_code_dir

---------

Co-authored-by: jiahy0825 <jiahongyu@baidu.com>

TimeYWL committed:
add full_ test case and update zlib

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=64046) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=64046) before we can accept your contribution.<br/><hr/>**ywl** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=64046) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=64046) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/Paddle?pullRequest=64046) before we can accept your contribution.<br/><hr/>**ywl** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/Paddle?pullRequest=64046) it.</sub>

SigureMo committed:
[Cleanup] Remove Flake8 config in `.editorconfig` (#64027)

enkilee committed:
【PIR Dist Op Reg No.19】 reg pull_box_sparse (#62982)

* fix

* fix

* fix

* fix

* fix

* fix

* add test

* add

* fix

* fix

* add out

* fix

* codestyle

* fix

* fix backward

* merge

gouzil committed:
[Dy2St][PIR] Hold backward program in GradNode (#63694)

Co-authored-by: xiongkun <xiongkun03@baidu.com>
Co-authored-by: Nyakku Shigure <sigure.qaq@gmail.com>

houj04 committed:
split test.cmake: add new test_cases.cmake (#64007)

risemeup1 committed:
[PIR] Support sparse_slice and sparse_sum in pt (#64009)

* support sparse_slice and sparse_sum in pt

* support sparse_slice and sparse_sum in pt

* support sparse_slice and sparse_sum in pt

tianshuo78520a committed:
option for WITH_CPP_TEST (#63896)

* option for WITH_CPP_TEST

* fix

* Fix

* Fix

SigureMo committed:
[PIR] Fix `attributes_num` of `SliceArrayOp` (#64013)

SigureMo committed:
[Dy2St] Use `full_graph=True` outside dy2st uts (part1) (#64058)

SigureMo committed:
[Dy2St] Use `full_graph=True` outside dy2st uts (part2) (#64059)

tc20042008 committed:
fix typo (#64060)

Co-authored-by: jiahy0825 <jiahongyu@baidu.com>

WintersMontagne10335 committed:
update (#64042)

co63oc committed:
Replace paddle/fluid/platform/device/gpu/gpu_dnn.h (#63819)

* Fix

* Fix

* Fix

co63oc committed:
Clean lookup_table_v2_op.h lookup_table_v2_op.cu (#64020)

* Fix

* ci

wanghuancoder committed:
refine GetTensorListFromArgs (#64045)

co63oc committed:
Revert "【Hackathon 6th Fundable Projects 3 No.60】Remove fluid operator chunk_…" (#64050)

This reverts commit 88b1a6ed30a6d66a0226a4152429981511659908.

Eddie-Wang1120 committed:
[Prim][PIR] support floor_divide op forward in prim pir (#64023)

* floor-div-dev

* update test

Hongqing-work committed:
[CINN] Reconstruct shape_analysis (#63790)

* reconstruct shape_analysis

* fix input value shape infer

* fix merge bugs

* fix concat and gather op InferSymbolicShape

* fix merge bug

* fix value_to_shape_or_data hash error and add some checks

* fix set shape for null value

* fix group op lazy infer

* add IsStaticShape check

* fix merge bug

* support static dim check and set for VectorType

* change auto to detail type

runzhech committed:
[XPU] fix bugs in processing of attention_mask and fix_seed_offset on XPU (#64003)

* [XPU] fix segmentfault caused by setting fix_seed_offset on XPU

* cast attention_mask to float32 when necessary

Hongqing-work committed:
fix merge bug (#64069)

SigureMo committed:
【Fix PIR Unittest No.125、147、481】Fix some 0D uts in PIR mode (part1) (#64064)

cyber-pioneer committed:
[Prim][VJP]support autogen to remove unused composite in .yaml (#64054)

* support autogen to remove unused composite in .yaml

* fix bug

SigureMo committed:
[PIR] Fix typo `set_pit_tests_properties` -> `set_pir_tests_properties` (#64063)

SigureMo committed:
[Dy2St] Use `full_graph=True` outside dy2st uts (part3) (#64066)

changeyoung98 committed:
[PIR save/load] Open more tests for paddle.save and paddle.load (#64044)

* open more tests for paddle.save and paddle.load

* fix

NKNaN committed:
API Improvement for paddle.nn.functional.group_norm and paddle.nn.GroupNorm (#63881)

* update group_norm

* update trt plugin

* update trt plugin

* fix trt plugin

* fix trt plugin

* fix test

* fix test

* fix ci windows inference

* update kernel function names and add v2 test

* fix

* fix fp16 test

co63oc committed:
Revert "【Hackathon 6th Fundable Projects 3 No.81】Remove fluid operators ctc_a…" (#64049)

This reverts commit 2134ead2500320b0101169a4809fb9f50c76cc77.

co63oc committed:
Clean paddle/fluid/operators/fused/attention_layer_norm.h (#64051)

* Fix

* Fix

co63oc committed:
Replace operators::math to phi::math in fluid/operators (#63854)

Aurelius84 committed:
[CINN]Clean usless loop_reorder_aligment tactic (#63998)

* [CINN]Clean usless loop_reorder_aligment tactic

* fix source

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.396】fluid operator yolo_box_head (#63783)

* Fix

* Fix

* Fix

* Fix

* Fix

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.240】fluid operator moe (#63929)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.82】fluid operator cudnn_lstm (#63936)

* Fix

* Fix

* Fix

* Fix

jiahy0825 committed:
[CINN] Remove useless log (#64052)

xiaoguoguo626807 committed:
[pir_save_load] add pir for test_jit_save_load.py (#63958)

* add jit load.train

* modify backward program lost

* modify

* combine eval and train

* modify 8 case of jit.save.load

* modify jit_save_load case

* rename jit_save_load

* change name all

* modify timeout

* modify new case

* modify TestJitSaveLoadMultiMethods

* modify cpu tensor no holder bug

kircle888 committed:
Flashattention support qkvpacked and varlen (#63289)

* Flashattention support qkvpacked and varlen

* fix codestyle

* fix codestyle

* FlashAttention kvReduceGQA Performance Optimization

* Fix problem with windows

* code clean

* update third_party/flashattn

* update errormsg and docs

* update api

* update doc

* update doctest

* update doc, test=document_fix

* update doc, test=document_fix

* Update python/paddle/nn/functional/flash_attention.py

Co-authored-by: zachary sun <70642955+sunzhongkai588@users.noreply.github.com>

* Update python/paddle/nn/functional/flash_attention.py

Co-authored-by: zachary sun <70642955+sunzhongkai588@users.noreply.github.com>

* update doc

---------

Co-authored-by: zachary sun <70642955+sunzhongkai588@users.noreply.github.com>

xingmingyyj committed:
【PIR Dist Op Reg No.20】 reg global_gather (#63867)

* reg global_gather

* reg global_gather

* reg_global_gather

* fix

* fix

* fix

* fix conflict

* fix conflict

* Update ops_api_gen.py

* Update ops_api_gen.py

0x45f committed:
Fix backward program kwargs error when process inplace value (#63939)

AndSonder committed:
【Hackathon 6th No.35】support kwargs for recompute when use_reentrant == True fix (#63880)

* support kwargs for recompute when use_reentrant == True

* recover third party

Aurelius84 committed:
[CINN]Add enable_fuse_parallel_matmul_pass flag for CINN (#64086)

* [CINN]Add enable_fuse_parallel_matmul_pass flag for CINN

* fix info

hxzd5568 committed:
[CINN] add symbolic arange (#64068)

* [CINN] add symbolic arange

* fix error in arange

tc20042008 committed:
Fix block free vars (#64077)

* fix typo

* refactor GetFreeVars(const pir::Block&)

* use exited util function to compute free vars of a block

---------

Co-authored-by: jiahy0825 <jiahongyu@baidu.com>

YuanRisheng committed:
【Fix PIR Unittest No.1-5】Fix test_sequence_mask in PIR mode (#64055)

* fix test_sequence_mask

* fix py3

* fix unittest

* fix py3

HarperCy committed:
[XPU] swiglu support xpu && xhpc date update && set value support bf16 (#64070)

co63oc committed:
Fix (#64092)

Aurelius84 committed:
[CINN]Open 7 cinn subgraph unitttest (#64097)

YKTian-x2b committed:
[paddle inference ,mmha]head_dim case 10/26 were deleted, because they would trigger misalignedAddr cudaError (#64081)

jiahy0825 committed:
[CINN] Fix conv2d infersymbolic (#64085)

SigureMo committed:
[PIR] Fix some ops with sub-block print formatting (#64089)

Aurelius84 committed:
[CINN]Support Dynamic Shape for CINN subgraph UT (#64076)

* [CINN]Support Dynamic Shape for CINN subgraph UT

* add more UT

* add prim flag

* fix flag

* fix atol

* fix timeout

* fix timeout

* fix sub_graph_3

kangguangli committed:
[BugFix] fix fused_conv2d_add_act_kernel when its inputs are NHWC (#63988)

* fix fused_conv2d_add_act_kernel when its inputs are NHWC

* Update fused_conv2d_add_act_kernel.cu

SigureMo committed:
【Fix PIR Unittest No.194、207、509】Fix some 0D uts in PIR mode (part2) (#64088)

tianshuo78520a committed:
Chang Failed approval (#64004)

ZelinMa557 committed:
fix DiagEmbedInferMeta when x_dims contains -1 (#63961)

Signed-off-by: ZelinMa557 <3388706467@qq.com>

SigureMo committed:
【Fix PIR Unittest No.428】Deprecate `test_block_rename_var` ut (#64078)

tc20042008 committed:
[CINN]Dump dim expr to py code (#64101)

* fix typo

* refactor GetFreeVars(const pir::Block&)

* use exited util function to compute free vars of a block

* dump pir dim_expr info into py code

---------

Co-authored-by: jiahy0825 <jiahongyu@baidu.com>

SigureMo committed:
[SOT] Fix stringify dtype object in guard string (#64118)

SigureMo committed:
[SOT] Fallback when breakgraph on comprehensive (#64126)

yulangz committed:
[CINN]Add InferSymbolicShape of fpn ops (#63947)

* Add DistributeFpnProposalsOpInferSymbolicShape;
Add RoiAlignOpInferSymbolicShape;
fix GatherOpInferSymbolicShape when axis input is tensor;
fix Expr set_type implementation of derived class;

* fix

* change shape_analysis to infer_context

* fix

ccsuzzh committed:
【Hackathon 6th Fundable Projects 4 No.6】remove async_executor (#63923)

* remove async_executor

* return ci

chen2016013 committed:
[CINN] Use Flag to open cinn in llama (#64074)

* update

* update

* update

xiaoguoguo626807 committed:
【Fix PIR Unittest No.135】fix test_io_save_load and test_jit_save_load (#64096)

* add jit load.train

* modify backward program lost

* modify

* combine eval and train

* modify 8 case of jit.save.load

* modify jit_save_load case

* rename jit_save_load

* change name all

* modify timeout

* modify new case

* modify TestJitSaveLoadMultiMethods

* modify cpu tensor no holder bug

* modify jit_save_load name

* move test_jit_save_load

* modify cmake

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.96】fluid operator detection_map (#63965)

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

ZelinMa557 committed:
[CINN] add constant fold for max/min dim expr (#63978)

* add test
* update test_infer_sym_shape_unary_op test

Signed-off-by: ZelinMa557 <3388706467@qq.com>

phlrain committed:
[CINN] use pass decomp flatten in dyshape  (#64115)

* disable flatten decomp in dyshape

* update

* revert unit test

phlrain committed:
[CINN]optimize reduce mean decomp (#63832)

* add unittest for reduce_mean

* close some test

* [CINN] Add Broadcast large shape unittest

* change file name

* update

* update

* remove usless code

* fix bug

* remove useless code

* update

---------

Co-authored-by: zyfncg <zhangyunfei07@baidu.com>
Co-authored-by: jiahongyu <jiahongyu@baidu.com>

danleifeng committed:
fix rpc_sync and rpc_async doc;test=develop (#64100)

co63oc committed:
Replace memory::Copy in fluid/operators (#64093)

* Fix

* Fix

co63oc committed:
Replace platform::XPUDeviceContext phi::XPUContext (#64065)

* Fix

* ci

co63oc committed:
Clean paddle/fluid/operators/fused/attn_feed_forward.h (#64062)

YuanRisheng committed:
【Fix PIR Unittest No.6-15】Fix book/* and part of test_comp_* in PIR mode (#64124)

* fix unittest

* fix unittest

* fix cmake

zxcd committed:
add type promotion for complex and real number. (#63842)

* add type promotion for complex and real number.

* fix

* reduce api support

* add more api support

* fix

* fix

* remove matmul

* add T+S logic.

* fix bug

* fix unittest

* fix

* fix

* fix unittest

* fix gumbel

* rm print

* fix more unittests.

* fix test_llama_group_log_softmax.py

* fix bug, and add 0-d + 0-d logic.

* rm print

* fix behavior of bool and int

* add unittest for all type promotion.

* rm unintest which is unsupport dtype

* fix

* fix

* add error unittest

* fix increase unittest

* bug fix

* fixed by comment

* remove useless code.

* fix

* fix

* fix TypePromotionForZeroDimTensor

* add inplace API support, add special case can skip type promotion (add x=float32,y=float16/bfloat16).

* add broatcast support for MultiPrecisionAddKernelImpl.

Hongqing-work committed:
[CINN] fix sub_graph_detector (#64135)

luotao1 committed:
Update CODEOWNERS test=document_fix (#64141)

jiahy0825 committed:
[CINN] Open performance unittest (#64134)

zhanglirong1999 committed:
[OneDNN][PIR] Add onednn_placement_pass (#63982)

* first commit of pass

* delete sys

* rename ut

* explicit class

* modify for comment

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.76】fluid operator crf_decoding (#63933)

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.294】fluid operator random_routing (#63928)

* Fix

* Fix

xingmingyyj committed:
Fix paddle.mode and paddle.bincount API (#63970)

* fix_infermeta

* Update binary.cc

* Update binary.cc

* Update binary.cc

* Update binary.cc

* Update binary.cc

* ci

0x45f committed:
Fix ParametersRecorder for processing stop_gradient (#64084)

* Fix ParametersRecorder for processing stop_gradient

* Fix cmake

zyfncg committed:
fix bug of symbol constraint (#64136)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.149-152】fluid operator fused_elementwise_add (#63996)

* Fix

* Fix

* Fix

* Fix

* Fix

hxzd5568 committed:
[CINN] Add symbolic select (#63971)

* fix where op infer meta check

* convert where to cinn select

* convert where to cinn select

* [CINN] add symbolic select for where

---------

Co-authored-by: phlrain <phliuhongyu@126.com>

changeyoung98 committed:
[Bug Fix] Fix bug in test_empty_like_op.py (#64073)

* fix bug

* fix bug

* add type for empty

xiaoguoguo626807 committed:
【Fix PIR JIT SaveLoad Unittest No.18】modify test_bert.py (#64151)

* modify test_bert.py

* ci bug

Hongqing-work committed:
delete redundant ShapeOptimizationPass (#64140)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.306】fluid operator requantize (#63772)

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

co63oc committed:
Replace framework::TensorCopy phi::Copy (#64131)

* Fix

* Fix

co63oc committed:
Clean paddle/fluid/operators/unique_op.h (#64156)

SigureMo committed:
[Infra] add codeowners for dy2st and codestyle (#64133)

changeyoung98 committed:
[PIR save/load]Open more tests (#64121)

* open more tests

* fix

* fix

* fix

GGBond8488 committed:
A template demonstrating how to test higher-order autodiff. (#62917)

* add composite high order test

* remove static

* add test subdirectory

* fix cmake error

* fix high grad test bug

* rm tanh

* add timeout limit

* add prim scope

* opt

* opt

* fix places

* opt

wentaoyu committed:
[DistDialect] Add pir reshard s to r (#63836)

* add s_to_r_reshard_func

fix conflicts

* remove debug log

* fix code style

cyber-pioneer committed:
[Prim][PIR] Decomp supports PIR Distribution (#64072)

* fix cmake download for quant data

* fix cmake

* pir distribution supports decomp forward

* fix code

* fix flag

* remove unused code

* fix code

* polish code

---------

Co-authored-by: houj04 <houj04@foxmail.com>

JZ-LIANG committed:
[AutoParallel-PIR] Remove program clone in Pass (#64137)

* remove clone

* revise typoes

* update unitest

heavyrain-lzy committed:
add reduce_avg op (#64120)

zeroRains committed:
[Prim] reduce_as op support uint8, in8, complex64  and complex128 (#63782)

* fix the include file

* add uint8 and int8

* suport complex64 and complex128

* fix docs

* fix conflict

* fix the third_party

zhangbo9674 committed:
[AutoParallel] Open sequence parallel opt (#64119)

* fix

* fix

cqulilujia committed:
[XPU] avoid malloc redundant memory when creating context in comm manager (#64139)

co63oc committed:
Clean fluid/operators/truncated_gaussian_random_op.h (#64155)

* Fix

* ci

co63oc committed:
Move paddle/fluid/operators/math/ to phi (#64159)

* Fix

* Fix

* Fix

Hongqing-work committed:
[CINN]fix InferSymbolicShape for builtin.slice and pd_op.stack (#64173)

yuanlehome committed:
Add remove_shadow_feed_pass for inference and op translater support assign_out_ (#64169)

* add remove_shadow_feed_pass for inference and op translater support assign_out_

* udpate

* fix ut

Eddie-Wang1120 committed:
update composite (#64025)

SigureMo committed:
[SOT] Update ut `test_06_call_function` to avoid type promotion error (#64170)

cyber-pioneer committed:
[Prim]polish prim FLAGS_prim_skip_dynamic (#64106)

* polish prim FLAGS_prim_skip_dynamic

* add backward blacklist

* fix bug

* fix special case

* fix test case

phlrain committed:
add send op side effect (#64200)

MarioLulab committed:
【Hackathon 6th No.50】 support backward for pylayer op (#63319)

* create manual_pylayer_op files and
add utility func

* add instance for py_callable,
add vjp handler in `call_vjp` interface

* add backward testcases and
fix bugs for segmentation error

* update testcases and fix bugs

* legacy_test/test_static_pylayer.py -> pir

* code polish

* pre-commit

* remove redundant combine op

* update dy2graph utils

* add dy2st support

* code polish

* phi::errors -> common::erros

* code polist

* pre-commit

* pre-commit

chen2016013 committed:
[CINN] Add rules for cinn_to_pd (#64152)

* add shape_ops_fallback_to_phi_pass

* fix test

* add ops for cinn_to_pd

* Update accuracy_check_pass.cc

* fix test_sub_graph_90 bug

* update

---------

Co-authored-by: zyfncg <zhangyunfei07@baidu.com>

winter-wang committed:
add dp precision demo for auto parallel. (#64144)

heavyrain-lzy committed:
【Distributed】Delete `scale_and_split_grads` function (#64154)

* delete scale_and_split_grads function

* fix image_classification_resnet.inference.model.tgz

HarperCy committed:
xpu rotary support use_neox_rotary_style == True (#64090)

ADream-ki committed:
【Hackathon 6th No.4】Add Ormqr API to Paddle -part (#63227)

* ormqr API

* update

* test

* update code style

* update ormqr api

* code_style

* update check

* update ormqr

* finish

* finish

* finish

* update code_style

* update

* finish

* update name

* update

* update information

* update

* update

* Update python/paddle/tensor/linalg.py

---------

Co-authored-by: zachary sun <70642955+sunzhongkai588@users.noreply.github.com>

xiaoguoguo626807 committed:
【pir save load 】modify Accuracy diff of Test bert (#64168)

co63oc committed:
Replace platform::CPUPlace phi::CPUPlace in fluid/operators (#63902)

* Fix

* Fix

* ci

* Fix

* ci

JiaWenxuan committed:
[CINN] Simplify broadcast while SetShapeOrDataForValue (#64148)

* simplify broadcast

* fix

* fix

* fix

zeroRains committed:
[Prim][PIR] reduce_as op brackward decomp (#64016)

* decomposite the backward of the  reduce_as op

* reduce_as is prim op and delete the composite config in yaml

jzhang533 committed:
remove tpm approval requirements in CI (#64212)

SigureMo committed:
[PIR] Mark more ops as no grad (#64204)

iosmers committed:
sharding stage1 V1 support Broadcast overlap Forward (#63945)

* sharding v1 overlap

* delete pybind

* add txt

* add b.txt

* delete test file

* add pybind

* add test case for stage1 v1 overlap

* add test case for stage1 v1 overlap

* update test case

* delete print optimizer

* update

* update models to layers

* update mlp1

xiaoyewww committed:
feat(pir): reg send_and_recv (#64203)

QingshuChen committed:
fix kunlun sharding v2 bug. Sharding v2 need inplace assign (#64209)

Co-authored-by: QingshuChen <Qingshu.Chen714@gamil.com>

onepick committed:
[DCU]Fix build error on rocm5.7 (#63738)

* [ROCM]Add patch for warprnnt/warpctc

* add rocm_path to cmake_prefix_path
* rm ambiguous apt because rocm has implemented

Signed-off-by: jiajuku <jiajuku12@163.com>

* [ROCM] fix build error on rocm5.7

* change expression of Xcompiler since clang17 only support
"-Xcompiler xxx" instead of "-Xcompiler=xxx"
* clang17 check warning mre strictly

Signed-off-by: jiajuku <jiajuku12@163.com>

* [ROCM] add dependent rocm lib

Signed-off-by: jiajuku <jiajuku12@163.com>

* [ROCM]fix build warning

rocm5 requires the include path: $ROCM_PATH/include

Signed-off-by: jiajuku <jiajuku12@163.com>

* clean code style issue

Signed-off-by: jiajuku <jiajuku12@163.com>

* Revert "[ROCM] fix build error on rocm5.7"

This reverts commit 2b3bb69002a5065a14541fdaf4e9391cfe9d5460.

* [ROCm]:support rocm 57

* change build option "-Xcompiler" expression, since clang17 reuqire
  "-Xcompiler xxx" instead of "-Xcompiler=xxx"
* remove Werror from HIP_HCC_FLAGS since clang17 check more strictly.

Signed-off-by: onepick <juku.jia@hygon.cn>

* Fix build error Xcompiler expression for clang17

Signed-off-by: onepick <jiauku@hygon.cn>

* Fix build warning

* sign-compare
* bitwise-instead-of-logical
* unknown-warning-option
* unused-lambda-capture
* unused-variable
* unused-but-set-variable
* reorder-ctor
* deprecated-copy-with-user-provided-copy
* unused-local-typedef
* missing-braces
* sometimes-uninitialized

Signed-off-by: onepick <jiauku@hygon.cn>

---------

Signed-off-by: jiajuku <jiajuku12@163.com>
Signed-off-by: onepick <juku.jia@hygon.cn>
Signed-off-by: onepick <jiauku@hygon.cn>
Co-authored-by: onepick <juku.jia@hygon.cn>
Co-authored-by: onepick <jiauku@hygon.cn>

cyber-pioneer committed:
Revert clone delete in distribution and add decomposed test case. (#64194)

* reset dist decomp test case

* Revert "[AutoParallel-PIR] Remove program clone in Pass (#64137)"

This reverts commit fd7c1ca933e926bd2ea6cf3a738dac6e4f3e9095.

* fix case

zyfncg committed:
fix bug of generate_shape infer_symbolic_shape (#64181)

jiahy0825 committed:
[CINN] Keep write+read global memory (#64214)

YKTian-x2b committed:
[Paddle Inference] GemmEpilogueOp with series of CUTLASS kernel (#61925)

SigureMo committed:
[Infra] Update codeowners to check dy2st ut only (#64231)

SigureMo committed:
[PIR] Use value in Executor `fetch_list` (part1) (#64235)

hxzd5568 committed:
[CINN] fix bug: remove redandunt conversion in select (#64217)

iosmers committed:
sharding stage1 V2 support Allgather overlap Forward (#64103)

* update sharding v2

* update allgather

* update

* fix code style

* add test case

* update case

* update

* update

---------

Co-authored-by: zhengzhonghui <zhengzhonghui@baidu.com>

enkilee committed:
【Fix PIR JIT SaveLoad Unittest No.19】 modify test_bmn.py (#64184)

* fix

* fix

* fix

* fix

co63oc committed:
Clean paddle/fluid/operators/detection/prior_box_op.h (#64230)

co63oc committed:
Replace include paddle/fluid/platform/device/xpu/xpu_header.h (#64229)

gouzil committed:
[ci] fix sot ci test commit check error (#64174)

Asthestarsfalll committed:
【Hackathon 6th Fundable Projects 1 1-1】Add `_typing` module to paddle (#63604)


---------

Co-authored-by: SigureMo <sigure.qaq@gmail.com>

SigureMo committed:
[PIR] Use value in Executor `fetch_list` (part2) (#64237)

yuanlehome committed:
inference support flash_attn (#64213)

* inference support flash_attn

* fix

* add ut

6clc committed:
cinn: compute inline after tile_first (#64172)

6clc committed:
cinn(test): uncomment type heap test (#64061)

SigureMo committed:
[Dy2St] Use `full_graph=True` in `test_jit_save_load` (part4) (#64243)

SigureMo committed:
[Dy2St] Update check approval allow list (#64242)

InsaneOnion committed:
Align the Assigner API code examples with the usage of ParamAttr in the ParamAttr API documentation. (#64232)

Aurelius84 committed:
[Prim]Support pow_grad/scale_grad/stack_grad vjp decomp rule (#64186)

* [Prim]Support pow_grad/scale_grad/stack_grad vjp decomp rule

* add UT

* fix get_mutable_attribute

* fix jinjia

* fix prim

* fix prim

* fix timeout

* add fixme

changeyoung98 committed:
[PIR saveload] Add _suffix_varname_dict for jit and open tests (#64239)

* open more tests

* fix

* fix

* fix

* open load state_dict from old format

huangjiyi committed:
[PIR] Add yield instruction (#64234)

* add yield_instruction

* rm CopyBranchOutput

* fix while_instruction

YuanRisheng committed:
[PIR]Add Dtype patch and fix test_tensor_scalar_type_promotion_dynamic in pir mode (#64187)

* fix test_tensor_scalar_type_promotion_dynamic in pir mode

* fix year

* fix py3

YuanRisheng committed:
fix prim (#64166)

dynamicheart committed:
[XPU] bump XHPC to 20240511 (#64163)

* [XPU] bump XHPC to 20240509

* update

co63oc committed:
Replace platform::GpuLaunchConfig in fluid/operators (#64236)

* Fix

* Fix

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.81】fluid operator ctc_align (#64102)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.172】fluid operator fusion_seqpool_cvm_concat (#64038)

* Fix

* Fix

* Fix

zhanglirong1999 committed:
[OneDNN] Fix ocr error since pass avx512 command (#64132)

* fix ocr avx error since pass avx512 command

* replace 512 core to 512f, limit restrict

* skip avx2 in new pir pass

* fix

phlrain committed:
[CINN]Remove duplicate cast (#64183)

* remove duplicate cast

* fix cast bug

co63oc committed:
Replace paddle/fluid/platform/for_range.h (#64260)

Aurelius84 committed:
[Prim]Support start/end index in prim decompose (#64258)

* [Prim]Support start/end index in prim decompose

* fix dst_vars

* fix comment

* remove fix_me

* fix end==start

tc20042008 committed:
Dump constraints to pir pycode (#64224)

* fix typo

* refactor hash and operator== for ShapeOrDataDimExprs

* dump constraints to pir pycode

* minor fix

---------

Co-authored-by: jiahy0825 <jiahongyu@baidu.com>

JZ-LIANG committed:
[AutoParallel-PIR] 3D HybridParallel Demo & Unitest (#64143)

* add 3d net

cyber-pioneer committed:
[Prim] add flag test cases and block prim vjp matmul by default (#64220)

* add flag test cases

* fix test case

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.76】fluid operator chunk_eval (#64108)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.368】fluid operator stft  (#64227)

* Fix

* Fix

* Fix

* Fix

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.397】fluid operator yolo_box_post (#63785)

ming1753 committed:
[Inference] add sparse_conv_optim_pass (#63067)

* add sparse_conv_optim_pass

Hongqing-work committed:
[CINN]add InferSymbolicShape for assign_out_ (#64249)

* add InferSymbolicShape for assign_out_

* add cstr

SigureMo committed:
[PIR] Fix flags description about pir (`new IR` -> `PIR`) (#64247)

enkilee committed:
【Error Message No. 3】	paddle/fluid/pir/dialect/operator/* (#62775)

* fix

* fix

* fix

* fix

* fix

* fix typo errors

* fix

* fix

* fix

* fix

* fix

* fix

pkuzyc committed:
[Auto Parallel] Support MoE expert parallelism in dygraph auto parallel (#63904)

* Support MoE in dygraph auto parallel

* modify the api name and add unit test

* remove partial_status-related functions in dist_attr

* add unit test to CMakeList

* add timeout property for ut

* add global_clip_norm in unit test

* remove print

* polish as comments

* bug fix

enkilee committed:
【Error Message No. 16】Replace CHECK_ in paddle/cinn/frontend/.cc,.h (#63157)

* fix

* fix

* fix

* fix

* fix

* fix

* fix

* fix

enkilee committed:
【Error Message No.11】Replace CHECK_ in paddle/cinn/adt/* (#63283)

* fix

* fix

* fix

* fix

danleifeng committed:
fix gpubox training (#63905)

* fix gpubox training;test=develop

runzhech committed:
[XPU] set the shape of tile's output when the input is 0D (#64279)

6clc committed:
cinn(test): fix test bug (#64275)

0x45f committed:
[PIR AMP]Fix opt.minimize error when no using grad scaler (#64284)

pkuzyc committed:
[DistDialect] Add pir nd_mesh reshard function (#64223)

* add nd_mesh_reshard_function for pir, partial_to_replicate part

* add s_to_r part in nd_mesh_reshard_func

* add infer local for all_gather op

* formalize the unit test and remove print code

* revert s_to_r unit test

* add cross_mesh unit test

* add pir_reshard to CI

* add pir_reshard to CI

* fix code style

zrr1999 committed:
[SOT][DyShape] add `ENV_SOT_ALLOW_DYNAMIC_SHAPE` (#64278)

SigureMo committed:
[PIR] Force decomp `matmul_grad` OP by use `FLAGS_prim_vjp_skip_default_ops=False` in `test_paddlescience` (#64286)

* [PIR] Force decomp `matmul_grad` OP by use `FLAGS_prim_vjp_skip_default_ops=False` in `test_paddlescience`

* use os.environ

* set env before import paddle

* add env to cmake

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.360】fluid operator sparse_attention(#64175)

zyfncg committed:
open the unittest of test_dyshape_mean (#64296)

co63oc committed:
Replace platform::Place in fluid/operators (#64250)

huangjiyi committed:
[PIR] Support -1 shape for loop_vars in WhileOp (#64272)

* support loopvars -1 shape

* rename var

* rename var

* update

LLee233 committed:
[PIR][oneDNN] Add shuffle_channel_detect_pass (#64053)

SigureMo committed:
[PIR] Mark `ShareData` as an inplace OP and fix inplace pass (#64195)

6clc committed:
cinn(op): optim compile time of select (#64263)

xiaoguoguo626807 committed:
【pir save load】 modify multi jit program parmeter name. test_jit_layer.py (#64281)

* modify name

* modify mkloutput

* modify ci

co63oc committed:
Clean operators/reduce_ops/reduce_op_xpu.h (#64269)

co63oc committed:
Clean plat = paddle::platform in fluid/operators (#64238)

co63oc committed:
Move elementwise_op_impl.cu.h to phi (#64259)

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

* Fix

WintersMontagne10335 committed:
[CINN]Add transform rules for cinn_op.reshape (#64303)

SigureMo committed:
【Fix PIR Unittest No.237】Fix ut `test_var_base` in PIR mode (#64308)

DongBaiYue committed:
[CINN][New Hardware Update] rename IRCudaSchedule (#64318)

* rename IRCudaScheduleInjective to IRGpuScheduleInjective
* rename IRCudaScheduleMatMul to IRGpuScheduleMatMul
* rename IRCudaScheduleReduce to  IRGpuScheduleReduce
* rename IRCudaScheduleBlockReduce to  IRGpuScheduleBlockReduce
* rename IRCudaScheduleBlockReduceInternal to IRGpuScheduleBlockReduceInternal
* rename IRCudaScheduleBlockShuffleReduce to IRGpuScheduleBlockShuffleReduce
* rename IRCudaTwoStepReduceSchedule to IRGpuTwoStepReduceSchedule

Hongqing-work committed:
[CINN]delete redundant InferSymbolicShape in passes (#64176)

* [CINN]delete redundant InferSymbolicShape in passes

* delete shape_optimization_pass.h in shape_analysis.h

* change name

* fix code style

JZ-LIANG committed:
[AutoParallel-PIR] Adapt SPMD Rule for LLaMA Hybrid Parallel - PartI (#64196)

* add 3d net

* enable 8 gpu unitest

* bugfix

* trigger

* update spmd for pir

* update spmd rule for reshape xshape

* bug fix for xshape

zxcd committed:
fix (#64302)

SigureMo committed:
[Infra] Fix description about numpy testing check (#64322)

SigureMo committed:
【Fix PIR Unittest No.511、533、542-544】Fix 5 ut in PIR mode (`test_pow`, `test_randn_op`, `test_transpose_op`, `test_flatten2_op`, `test_einsum_op`) (#64307)

xiaoguoguo626807 committed:
【pir_save_load】support pir save of test_custom_relu_model.py (#64327)

* add test_custom_relu_model.py

* add test_custom_relu_model.py

NKNaN committed:
【Hackathon 6th No. 7】Add sinc / sinc_ API to Paddle -part (#63521)

* add sinc

* fix float16

* remove cpu fp16 test

* fix tolerance

* update test

* fix timeout

* fix fp16 test

* fix fp16 test

* fix fp16 test

* add dtype check

* update check dtype

* update test

* fix init

* add bfloat16 test and fix underflow problem

* rerun ci

* fix test

* fix test

* fix bf16

* update test

* update test

* update test

* rerun ci

* fix bf16 test

* rerun ci

* add gradient test

* add gradient test

Aurelius84 committed:
【Fix PIR Unittest No.213、277】Support to decompose squeeze_grad and unsqueeze_grad (#64277)

* [Prim]Add more detail note for start/end index test=document_fix

* migrate UT

* rename UT

* fix timeout

* fix atol

co63oc committed:
Fix (#64339)

co63oc committed:
Clean paddle/fluid/operators/unbind_op.h (#64338)

* Fix

* ci

co63oc committed:
Fix (#64336)

co63oc committed:
Replace gpu/gpu_info.h in fluid/operators (#64335)

* Fix

* ci

zrr1999 committed:
[SOT] use better  Singleton (#64340)

co63oc committed:
Clean paddle/fluid/operators/squeeze_op.h(#64337)

zhangbo9674 committed:
Refine swiglu infer_spmd rules (#64301)

* add

* add

megemini committed:
【Hackathon 6th No.13】为 Paddle 新增 RAdam / NAdam API -part (#63671)

* add nadam cpu op

* test nadam cpu op

* add nadam gpu op

* add nadam docstring

* mod nadam docstring

* add radam op

* fix conflict

* codestyle

* fix & add unittest

* remove momentum_decay_base

* [Change] accumulator scalar to tensor & add unittest

* remove glog

* make test lr smaller

* make test lr smaller

* nadam & radam test timeout

* use fake test data instead of uci data

* [Update] change nadam lr to 0.002

* [Update] docstring

* [Update] unittest

SigureMo committed:
[PIR] Refine code about inplace pass (#64317)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.23】fluid operator box_clip  (#64028)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.260】fluid operator p_send (#64287)

* Fix

* Fix

* ci

zhangbo9674 committed:
Add fuse all_reduce_sum with split to reduce_scatter pass (#64316)

* add

* fix

* add

* add

* add ut

* add ut

* add ut

wanghuancoder committed:
【Fix PIR Unittest No.65,67,68,69,70,163,262,293】Fix some test cast in PIR mode (#64314)

* fix pir api 2

SigureMo committed:
[PIR] Add Common Subexpression Elimination pass (#63960)

winter-wang committed:
move mix2dist pass to python end. (#64299)

dynamicheart committed:
[XPU] remove cast for elementwise_add(float, bf16/fp16) (#64289)

* [XPU] remove cast for elementwise_add(float, bf16/fp16)

* fix

* fix

NKNaN committed:
API Improvement for paddle.nn.initializer.TruncatedNormal 易用性提升 (#64110)

* update truncated_gaussian_random kernel

* update docs

* fix error

* fix opp_version

* fix build error

* fix test

* add note in en docs and fix xpu test

* fix xpu test

* fix xpu test

* update

* update fluid

* rm fluid operator

* fix xpu test

* fix xpu test

* update note

jeng1220 committed:
bugfix, add skipIf for multigpu tests (#62499)

* bugfix, add skipIf for multigpu tests

* Bugfix test_op with single GPU

NKNaN committed:
【Hackathon 6th No.53】move fake_channel_wise_quantize_abs_max/fake_channel_wise_quantize_dequantize_abs_max op to phi - part (#64266)

* move fake_channel_wise_quantize_abs_max and fake_channel_wise_quantize_dequantize_abs_max to phi

* fix opps yaml

* fix ops yaml

* fix accuracy

AndSonder committed:
【Hackathon 6th No.34】support return micro batch loss for dygraph train_batch (#64218)

* support return micro batch loss

* fix codestyle

* recover some code

Vvsmile committed:
refine the loop_transformation strategy to remove the extra if condition for exact division case (#63140)

YibinLiu666 committed:
Support exclusive and reverse for cumprod (#64022)

* add reverse and exclusive

* support reverse and exclusive

* fix inplace test

* fix make on xpu

* remove print

* update test time

* add 1-5D test

* speed up test

* update test

* Update CMakeLists.txt

* update op_version

houj04 committed:
fp16 and bf16 support for CPU GaussianKernel (#64304)

* fp16 and bf16 support for CPU GaussianKernel

* fix ut.

wanghuancoder committed:
【Fix PIR Unittest No.60-64】Fix some test cast in PIR mode (#64276)

* fix pir api 1

co63oc committed:
Clean paddle/fluid/operators/expand_as_v2_op.h (#64349)

xiaoguoguo626807 committed:
【Fix PIR JIT SaveLoad Unittest No.20-21】modify test_mobile_net.py， test_resnet.py (#64315)

* modify test_modbile_net.py

* delete print

* modify ci

* code style

* code style

zeroRains committed:
fix the bug in sigmoid_cross_entropy_with_logits_grad_kernel (#64253)

* fix the bug in sigmoid_cross_entropy_with_logits_grad_kernel

* add the test for sigmoid_cross_entropy_with_logits_grad

* modify the test

* modify the grad  kernel

* modify the kernel

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.90】fluid operator depend (#64206)

phlrain committed:
add index sample infer symbol (#64332)

co63oc committed:
【Hackathon 6th Fundable Projects 3 No.109】Remove fluid operator dropout_nd (#63480)
